grails-app/services/com/unifina/service/ProductService.groovy

```diff
@@ -13,6 +13,20 @@ class ProductService {
 	PermissionService permissionService
 	SubscriptionService subscriptionService
 
+	void removeUsersProducts(String username) {
+		def query = Product.where {
+			owner == username
+		}
+		def all = query.findAll()
+		all.each{ it ->
+			Product product = (Product) it
+			product.streams.toArray().each{ Stream stream ->
```

Comment:
Are you sure the streams need to be deleted manually? I think there is a GORM feature that will auto-delete associations?

Reply:
I wanted to make sure that callback in removeStreamFromProduct() gets called, but I'll drop it.